site-scrapers/BostonMedicalCenter/index.js

```diff
@@ -0,0 +1,57 @@
+const { sites } = require("./config");
+const mychart = require("../../lib/MyChartAPI");
+
+module.exports = async function GetAvailableAppointments() {
+    console.log("BMC starting.");
+    const webData = await ScrapeWebsiteData();
+    console.log("BMC done.");
+    const results = [];
+    const timestamp = new Date();
+
+    for (const site of sites) {
+        results.push({
+            ...webData[site.departmentID],
+            ...site,
+            timestamp,
+        });
+    }
+    return results;
+};
+
+async function ScrapeWebsiteData() {
+    // We need to go through the flow and use a request verification token
+    const [
+        cookie,
+        verificationToken,
+    ] = await mychart.GetCookieAndVerificationToken(
+        "https://mychartscheduling.bmc.org/mychartscheduling/SignupAndSchedule/EmbeddedSchedule?id=10033319,10033364,10033367,10033370,10033373&dept=10098245,10098242,10098243,10098244,10098241&vt=2008&lang=en-US"
```

Comment:
We shouldn't have identical static data in multiple places (department IDs), and we should try to have put similar data in similar places (provider ids and department ids).

Is this perhaps

```
"https://mychartscheduling.bmc.org/mychartscheduling/SignupAndSchedule/EmbeddedSchedule" +
"?id="+
sites.map( s => s.providerID).join(",")+
"&dept="+
sites.map( s => s.departmentID).join(",")+
"&vt=2008&lang=en-US";
```

after adding `providerID`s to `sites`?

Oh wait — you already do something similiar with `const deptList` on L51, so…please refactor this to do it once in one place, a helper function or a top-level const or whatever.

---

I was tempted to suggest being fancy with
```
let u = new URL('https://mychartscheduling.bmc.org/mychartscheduling/SignupAndSchedule/EmbeddedSchedule');
u.search = new URLSearchParams( {
  id: sites.map( s => s.myChartId).join(",")
...
```

but it turns out that URLSearchParams has weird escaping rules that are unfriendly:
```
> new URLSearchParams({ 'id': '1,2,3' }).toString()
'id=1%2C2%2C3'
```

Reply:
Yeah good call I can do that. 

For context, most of the other MyChart scrapers have all this hard coded, but they also don't care about the department IDs because they're only getting info for a single site (this one and UMassMemorial use the department IDs to separate availability by site).